frontend_tests/zjsunit/i18n.js

```diff
@@ -1,5 +1,59 @@
 "use strict";
 
+const {createIntl, createIntlCache} = require("@formatjs/intl");
+const _ = require("lodash");
+
+const cache = createIntlCache();
+
+exports.intl = createIntl(
+    {
+        locale: "en",
+        defaultLocale: "en",
+        defaultRichTextElements: Object.fromEntries(
+            ["b", "code", "em", "i", "kbd", "p", "strong"].map((tag) => [
+                tag,
+                (content_html) => `<${tag}>${content_html}</${tag}>`,
+            ]),
+        ),
+    },
+    cache,
+);
+
+exports.$t = (descriptor, values) =>
+    "translated: " +
+    exports.intl.formatMessage(
+        {
+            ...descriptor,
+            id: `${descriptor.defaultMessage}#${descriptor.description}`,
+        },
+        values,
+    );
+
+const default_html_elements = Object.fromEntries(
+    ["b", "code", "em", "i", "kbd", "p", "strong"].map((tag) => [
+        tag,
+        (content_html) => `<${tag}>${content_html}</${tag}>`,
+    ]),
+);
+
+exports.$t_html = (descriptor, values) =>
+    "translated HTML: " +
+    exports.intl.formatMessage(
+        {
+            ...descriptor,
+            id: `${descriptor.defaultMessage}#${descriptor.description}`,
+        },
+        {
+            ...default_html_elements,
+            ...Object.fromEntries(
+                Object.entries(values ?? {}).map(([key, value]) => [
+                    key,
+                    typeof value === "function" ? value : _.escape(value),
+                ]),
+            ),
+        },
+    );
```

Comment:
Do we want to add node tests for this function?

Reply:
This function *is* a Node test. The corresponding function in `static/js/i18n.js` is covered by `frontend_tests/node_tests/i18n.js`, although that coverage temporarily disappears in the interior of this branch until the template helpers are switched.

Reply:
I see.  So I guess what I actually want here is some way to deduplicate this from the non-test version, but that can wait.